src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -128,4 +128,26 @@ partial class BoundThrowExpression
     {
         protected override ImmutableArray<BoundNode> Children => ImmutableArray.Create<BoundNode>(this.Expression);
     }
+
+    internal abstract partial class BoundMethodOrPropertyGroup
+    {
+        protected override ImmutableArray<BoundNode> Children => ImmutableArray.Create<BoundNode>(this.ReceiverOpt);
+    }
+
+    internal partial class BoundSequence
+    {
+        protected override ImmutableArray<BoundNode> Children => StaticCast<BoundNode>.From(this.SideEffects.Add(this.Value));
+    }
+
+    internal partial class BoundStatementList
+    {
+        protected override ImmutableArray<BoundNode> Children
+        {
+            get
+            {
+                System.Diagnostics.Debug.Assert(this.Kind == BoundKind.StatementList || this.Kind == BoundKind.Scope);
```

Comment:
Will this assert fail when evaluating the property in the debugger?

Reply:
let me try. didn't thought of that...

Reply:
yes. it does. it shows 
"Children = 'boundBlock.Children' threw an exception of type 'Microsoft.CodeAnalysis.ThrowingTraceListener.DebugAssertFailureException"
in debug window. 

it used to show empty. now that exception for BoundBlock, and child statements for other such as StatementList or Scope.

Reply:
Consider removing the assert.

Reply:
but we dont want BoundBlock.Children to be used.

Reply:
An assert in a property may result in a popup in the debugger though. If it's important to avoid `BoundBlock.Children`, perhaps return `ImmutableArray<BoundNode>.Empty` when `Kind` is not one of the expected values.

Reply:
sure I can do that.

PS: it doesn't cause pop up, but value in the watch window or data tip says assert exception was thrown.

Reply:
Aren't statements inside a block children of that block? I think they are.

Reply:
Children should return something if the bound node is not supported by IOperation yet. this is a workaround we have until we can support all bound nodes we want to support. BoundBlock is already supported so children should never be called. that's why I have put assert.